plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageService.java

```diff
@@ -162,10 +170,37 @@ StorageOptions createStorageOptions(final GoogleCloudStorageClientSettings clien
         }
         if (Strings.hasLength(clientSettings.getProjectId())) {
             storageOptionsBuilder.setProjectId(clientSettings.getProjectId());
+        } else {
+            String defaultProjectId = null;
+            try {
+                defaultProjectId = ServiceOptions.getDefaultProjectId();
```

Comment:
Did we allow plugins to read/write any system env variable? (I don't fully remember)

If so, maybe this will need some adaptation for FIPS JVM like what is done for EC2 (see #52046)?

Reply:
We only read the environment variable, which is allowed by policy.
#52046 is unrelated to what we're doing here (as we're not writing a system property in a test)

Reply:
Oh, sorry for the noise :facepalm: